SH4RE/SH4RE/Login/CreateAccountView.swift

```diff
@@ -29,123 +33,134 @@ struct CreateAccountView: View {
+                                Button("Photo Library", action: {showPhotoLibSheet.toggle()})
+                            } label: {
+                                VStack {
+                                    Image(uiImage: profilePicture)
+                                        .resizable()
+                                        .clipShape(Circle())
+                                        .aspectRatio(contentMode: .fill)
+                                        .frame(width: screenSize.width * 0.1, height: screenSize.height * 0.1)
+                                    Image(systemName: "camera.fill")
+                                        .foregroundColor(.primaryDark)
+                                }
                             }
-                            else if (password != confirmPassword) {
-                                errorInField = true
-                                errorDescription = "Passwords did not match"
+                            .sheet(isPresented: $showPhotoLibSheet) {
+                                ImagePicker(sourceType: .photoLibrary, selectedImage: $profilePicture)
                             }
-                            else {
-                                Task {
-                                    do {
-                                        try await Auth.auth().createUser(withEmail: username, password: password)
-                                        currentUser.hasLoggedIn = true
-                                        documentWrite(collectionPath: "User Info", uid: Auth.auth().currentUser!.uid, data: ["name": name,"email": username])
-                                    }
-                                    catch {
-                                        errorInField = true
-                                        errorDescription = error.localizedDescription
-                                    }
-                                }
+                            .sheet(isPresented: $showCameraSheet) {
+                                ImagePicker(sourceType: .camera, selectedImage: $profilePicture)
                             }
-                        })
-                        {
-                            Text("Create Account")
+                            Text("Name")
+                                .font(.system(size: 18))
+                                .frame(maxWidth: screenSize.width * 0.8, alignment: .leading)
+                            TextField("Your Name", text: $name)
+                                .disableAutocorrection(true)
+                                .frame(width: screenSize.width * 0.8)
+                                .textFieldStyle(textInputStyle())
+                                .padding(.bottom)
+                            Text("Email Address")
+                                .font(.system(size: 18))
+                                .frame(maxWidth: screenSize.width * 0.8, alignment: .leading)
```

Comment:
if you align to . leading, you shouldn't need to put a frame eon all the text. In order to center the photo after making everything leading, you can put the profile photo in a HStack with spacers on both sides, should remove a ton of these frames and manual sizing. Also, something that is really helpful as well is the .fixedSize modifier. Might help in the future